front/app/containers/AggDropDown/AggDropDown.tsx

```diff
@@ -137,7 +134,10 @@ interface AggDropDownState {
   filter: string;
   buckets: AggBucket[];
   prevParams: SearchParams | null;
-  sort: boolean;
+  alphaDesc: boolean;
+  numericDesc: boolean;
+  alphaActive: boolean;
+  numberActive: boolean;
```

Comment:
After attempting to refactor the code to incorporate typing, I have found that maybe the booleans were a necessary evil, unless you have an idea of how to deal with this problem:

The icons that the user interacts with to change both the direction (asc, desc)  and the type (alpha, number) of the sort are managed by a stateless component that takes in different types. 

`<Sorter type='alpha' asc={alphaDesc} isActive={alphaActive} toggle={this.toggleAlphaSort}/>
          <Sorter type='number' asc={numericDesc} isActive={numberActive} toggle={this.toggleNumericSort}/>`

With separate active and sort direction states, I was able to have the icons render according to what was last clicked. So, if you were toggling the alpha sort, the number sort icon would stay static.

I got the components to work with the enum instead of the bool for which sort is active, along with a singular sort direction boolean in state. This has created a behavior, however, that causes both icons to change directions when descending is toggled in state. It is a little visually jarring, but still functional. I haven't been able to figure out a way to have them toggle independently of one another with the way the Sorter components are designed right now. I could make individual components for numeric and alpha sort, but I feel that is even more repetitive.

I'm going to commit what I have. If you can take a look and give me feedback, I would appreciate it.

Reply:
I don't see the commit yet and can look once you've pushed it.
In this case 
`<Sorter isActive={alphaActive} />` 
might be replaced by 
`<Sorter isActive={sort == SortKind.Alpha} />`

Reply:
This helped me figure it out! Thanks, about to add a new commit with changes.